### ISSUE TITLE
Dev/Peer Dependencies with Bitsrc components

### ISSUE THREAD

**Issue by michaelryancaputo** [teambit/bit#1137]
i'm creating a series of components that use existing npm components to render, they happen to use styled-components.

When I export my bitsrc component and import it into my main app, I am getting a message: `It looks like there are several instances of 'styled-components initialized in this application`
In the past, i've used devDepenencies, and peerDepenencies for this type of package to avoid those components from exporting the styled-components library with the package.

There seems to be some documentation about how to use devDependencies with bitsrc packages. But it doesn't actually explicitly say how I could achieve that, and export them to avoid this message.

## Expected Behavior
- I would expect that my bitsrc components do not include a copy of devDepenencies, or peerDepenencies.
- My application should not give me a message `It looks like there are several instances of 'styled-components initialized in this application`

## Actual Behavior
- I am getting a message `It looks like there are several instances of 'styled-components initialized in this application`

## Steps to Reproduce the Problem

  1. create bitsrc component that uses a npm package that uses styled-components
  2. export that component that already has styled-components installed
  3. render that component

## Specifications

  - Bit version: 13.0.3
  - Node version: 8.11.1
  - npm / yarn version: yarn@1.7.0
  - Platform: OSX
  - Bit compiler (include version): bit.envs/compilers/react@0.0.13
  - Bit tester (include version): bit.envs/testers/jest@0.0.32

**Comment by GiladShoham** [comment]
Hi @michaelryancaputo ,

**TL;DR**

All you have to do is set styled-components under peerDependencies in the project's package.json then tag and export the components.

### More details about the process

(I'll make sure to update the docs, as they are not as granular).

Dependencies, devDependencies, and peerDependencies are calculated automatically by Bit. Bit than takes the results, and create a package.json file of the Component.

Here's a quick recap of how it works:

Bit reads each component's code AST and lists all required/imported packages. Afterward, it splits those packages into three categories; dependencies, devDependencies, and peerDependencies using the following set of rules:
1. If the required package is configured in the root package.json as a peerDependency, Bit will set it as a peerDependency for all components that depends on it. 
2. According to the specific file-type within the Component (implementation or a test-file) that requires a package; 
*  Bit will set a devDependency for all packages required by test files.
*  A dependency will set for all packages required by implementation-files.

Several notes:
* If the same package is listed as a peerDependency and a dependency/devDependency in your package.json, Bit will consider it as a   peerDependency.
* Package required by both test-file and implementation-file will be set as a dependency.

Hope that makes sense for you.
Gilad.

**Comment by modimation** [comment]
peerDependency mapping does not appear to working:

I have the following peer and dev dependencies mapped in my package.json file:

`{
  "name": "@hivekit/button",
  "version": "0.2.59",
  "description": "A simple button component",
  "main": "dist/index.js",
  "scripts": {
    "buildLocal": "../../build/scripts/prepublish-with-babel.sh button",
    "prepublish": "../../node_modules/.bin/in-publish && ../../build/scripts/prepublish-with-babel.sh button || ../../node_modules/.bin/not-in-publish"
  },
  "keywords": [
    "Button"
  ],
  "author": "tnb",
  "license": "ISC",
  "dependencies": {
    "babel-runtime": "6.26.0",
    "styled-system": "2.0.2",
    "what-input": "5.0.5"
  },
  "peerDependencies": {
    "react": ">=16.4.0",
    "react-dom": ">=16.4.0",
    "styled-components": ">=3.2.3"
  },
  "devDependencies": {
    "react": ">=16.4.0",
    "react-dom": ">=16.4.0",
    "styled-components": ">=3.2.3"
  }
 ...
}`

The package.json file generated for the Bit components exported to remote scope:

`{
  "name": "@bit/hivekit.components.button",
  "version": "0.2.60",
  "homepage": "https://bitsrc.io/hivekit/components/button",
  "main": "dist/index.js",
  "dependencies": {
    "prop-types": "15.6.1",
    "react": "16.4.0",
    "styled-components": ">=3.2.3",
    "styled-system": "2.0.2",
    "what-input": "5.0.5"
  },
  "devDependencies": {
    "jest-styled-components": "5.0.0"
  },
  "peerDependencies": {},
  "license": "SEE LICENSE IN LICENSE"
}`

Based on the comments above "styled-components": ">=3.2.3" should be listed as a  peerDependency but instead is listed as a dependency.

**Comment by GiladShoham** [comment]
@modimation 
which version of bit are you using?
I just copied your package.json and create a new file with this code:
```js
require('react');
require('react-dom');
require('styled-components');
```
Then export and import the component to another project
All the dependencies are peer dependencies as expected after importing.

Maybe you can share your require statements?

Something else which might cause this is if you have a closer package.json than the one in the root of your workspace.
If for example you have a package.json in the component folder (in the workspace you tagged the component), this package.json will have a priority over the one in the workspace root.
